diff -ru SimDataFormats/ValidationFormats/interface/MaterialAccountingTrack.h SimDataFormats/ValidationFormats/interface/MaterialAccountingTrack.h
--- a/SimDataFormats/ValidationFormats/interface/MaterialAccountingTrack.h	2019-12-06 19:49:51.346449757 +0100
+++ b/SimDataFormats/ValidationFormats/interface/MaterialAccountingTrack.h	2019-12-06 19:50:28.693116122 +0100
@@ -3,7 +3,7 @@
 
 #include <vector>
 #include "DataFormats/GeometryVector/interface/GlobalPoint.h"
-#include "G4VPhysicalVolume.hh"
+#include "Geant4/G4VPhysicalVolume.hh"
 #include "SimDataFormats/ValidationFormats/interface/MaterialAccountingStep.h"
 #include "SimDataFormats/ValidationFormats/interface/MaterialAccountingDetector.h"
 
