--- DDDigi/src/DigiKernel.cpp	2020-03-06 19:53:02.999992067 +0100
+++ DDDigi/src/DigiKernel.cpp	2020-03-06 19:51:51.799992404 +0100
@@ -88,15 +88,15 @@
 class DigiKernel::Wrapper  {
 public:
   DigiContext& context;
-  DigiEventAction*  action = 0;
-  Wrapper(DigiContext& c, DigiEventAction* a)
+  DigiEventAction&  action;
+  Wrapper(DigiContext& c, DigiEventAction& a)
     : context(c), action(a) {}
   Wrapper(Wrapper&& copy) = default;
   Wrapper(const Wrapper& copy) = default;
   Wrapper& operator=(Wrapper&& copy) = delete;
   Wrapper& operator=(const Wrapper& copy) = delete;
   void operator()() const {
-    action->execute(context);
+    action.execute(context);
   }
 };
 
