--- DDDigi/CMakeLists.txt	2019-10-25 15:38:41.000000000 +0200
+++ DDDigi/CMakeLists.txt	2019-10-29 14:13:42.163703318 +0100
@@ -28,8 +28,8 @@
   dd4hep_print( "|++> TBB_LIBRARY     -->  ${TBB_LIBRARY}")
   dd4hep_print( "|++> TBB found. DDDigi will run multi threaded.")
   target_compile_definitions(DDDigi PUBLIC DD4HEP_USE_TBB)
-  target_link_libraries(DDDigi ${TBB_LIBRARY})
-  target_include_directories(DDDigi ${TBB_INCLUDE_DIRS})
+  target_link_libraries(DDDigi PUBLIC ${TBB_LIBRARY})
+  target_include_directories(DDDigi PUBLIC ${TBB_INCLUDE_DIR})
 else()
   dd4hep_print( "|++> TBB not found. DDDigi will only work single threaded.")
 endif()
