--- /home/jonas/tmp/PKGBUILDs/dd4hep/src/DD4hep-01-11/DDDigi/src/DigiKernel.cpp.bak	2019-10-29 14:38:31.317024561 +0100
+++ /home/jonas/tmp/PKGBUILDs/dd4hep/src/DD4hep-01-11/DDDigi/src/DigiKernel.cpp	2019-10-29 14:39:44.007023918 +0100
@@ -91,7 +91,7 @@
   DigiAction*  action = 0;
   Wrapper(DigiContext& c, DigiAction* a)
     : context(c), action(a) {}
-  Wrapper(Wrapper&& copy) = delete;
+  Wrapper(Wrapper&& copy) = default;
   Wrapper(const Wrapper& copy) = default;
   Wrapper& operator=(Wrapper&& copy) = delete;
   Wrapper& operator=(const Wrapper& copy) = delete;
@@ -111,7 +111,7 @@
   DigiKernel& kernel;
 public: 
   Processor(DigiKernel& k) : kernel(k) {}
-  Processor(Processor&& l) = delete;
+  Processor(Processor&& l) = default;
   Processor(const Processor& l) = default;
   void operator()()  const {
     int todo = 1;
